src/main/java/nl/unimaas/bigcat/wikipathways/nanopubs/PubMed.java

```diff
@@ -0,0 +1,96 @@
+package nl.unimaas.bigcat.wikipathways.nanopubs;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+public class PubMed {
+	   public static Date getPublicationDate(String pubmedId) throws IOException, ParseException {
+		   if(isNumeric(pubmedId))
+		   {
+			    String postUrl = "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=pubmed&id="+pubmedId+"&retmode=json"; //post url
+			    //URLConnection conn = null;
+			    
+//	    		URL url = new URL(postUrl);
+//		        conn = url.openConnection();
+//		        conn.setRequestProperty("User-Agent", "Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2228.0 Safari/537.36");
+		       
+			    URL url = new URL(postUrl);
+			    HttpURLConnection conn = (HttpURLConnection)url.openConnection(); 
+			    conn.setRequestMethod("GET");
+			    conn.connect();
+			    int responsecode = conn.getResponseCode();
+			    if(responsecode != 200)
+			    {
+			    	System.out.println(pubmedId);
+			    	throw new RuntimeException("HttpResponseCode: " +responsecode);
+			    }
+			    else
+			    {
+			    	InputStreamReader inputStreamReader = new InputStreamReader(conn.getInputStream());
+			    	BufferedReader bufferedReader = new BufferedReader(inputStreamReader);
+			    	JSONParser parse = new JSONParser();
+			    	JSONObject jobj = (JSONObject)parse.parse(bufferedReader);
+			    	JSONObject resultJson = (JSONObject) jobj.get("result");
+			    	JSONObject pubmedJson = (JSONObject) resultJson.get(pubmedId);
+			    	String jsonDate = "";
+			    	
+			    	conn.disconnect();
+			    	
+			    	if (!((String)pubmedJson.get("epubdate")).equals(""))

```

Comment:
When I run the code, I get this exception:

```
Exception in thread "main" java.lang.NullPointerException
        at nl.unimaas.bigcat.wikipathways.nanopubs.PubMed.getPublicationDate(PubMed.java:48)
        at nl.unimaas.bigcat.wikipathways.nanopubs.NanoPubs.updateNanopublication(NanoPubs.java:147)
        at nl.unimaas.bigcat.wikipathways.nanopubs.NanoPubs.createNanoPublications(NanoPubs.java:103)
        at nl.unimaas.bigcat.wikipathways.nanopubs.ComplexesPubs.main(ComplexesPubs.java:32)
```

Reply:
I hope you will not get this exception now.